sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs

```diff
@@ -281,6 +290,38 @@ public void Notify()
             }
         }
 
+        internal int GetMessageReceiveCount(out TaskSeriesCommandResult delay)
+        {
+            delay = default(TaskSeriesCommandResult);
+            int numMessagesToReceive = _queueProcessor.QueuesOptions.BatchSize;
+
+            if (_concurrencyManager != null && _concurrencyManager.Enabled)
+            {
+                // When not using DC, WaitForNewBatchThreshold cleans up tasks as they complete.
+                // When using DC we need to also do cleanup of completed tasks, particularly
+                // before we use the count of pending tasks in the calculation below.
```

Comment:
Is having correct count the only goal? If so can we do `_processing.Count(p => !p.IsCompleted)` and keep the cleanup in a single place?

Reply:
It's not just to have a correct count, it's also to remove completed tasks. When DC is enabled, the other cleanup logic [here](https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs#L306) isn't running.